lib/devise/models/recoverable.rb

```diff
@@ -33,6 +33,8 @@ def self.required_fields(klass)
       # Update password saving the record and clearing token. Returns true if
       # the passwords are valid and the record was saved, false otherwise.
       def reset_password(new_password, new_password_confirmation)
+        self.errors.add(:password, :blank) && return unless new_password
```

Comment:
About the `.blank?` the validations take care of it automatically (that's why the original `.to_s` worked). Should I still add it to make it more explicit?